src/main/java/org/candlepin/subscriptions/subscription/ExportSubscriptionListener.java

```diff
@@ -0,0 +1,163 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.subscription;
+
+import com.redhat.cloud.event.apps.exportservice.v1.Format;
+import com.redhat.cloud.event.apps.exportservice.v1.ResourceRequest;
+import com.redhat.cloud.event.apps.exportservice.v1.ResourceRequestClass;
+import com.redhat.cloud.event.parser.ConsoleCloudEvent;
+import com.redhat.cloud.event.parser.ConsoleCloudEventParser;
+import com.redhat.swatch.clients.export.api.client.ApiException;
+import com.redhat.swatch.clients.export.api.model.DownloadExportErrorRequest;
+import com.redhat.swatch.clients.export.api.resources.ExportApi;
+import jakarta.ws.rs.core.Response.Status;
+import java.util.List;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.candlepin.subscriptions.exception.ExportServiceException;
+import org.candlepin.subscriptions.rbac.RbacApiException;
+import org.candlepin.subscriptions.rbac.RbacService;
+import org.candlepin.subscriptions.task.TaskQueueProperties;
+import org.candlepin.subscriptions.util.KafkaConsumerRegistry;
+import org.candlepin.subscriptions.util.SeekableKafkaConsumer;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Profile;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.stereotype.Service;
+
+/** Listener for Export messages from Kafka */
+@Service
+@Slf4j
+@Profile("capacity-ingress")
+public class ExportSubscriptionListener extends SeekableKafkaConsumer {
+  private final ConsoleCloudEventParser parser;
+
+  private static final String ADMIN_ROLE = ":*:*";
+  private static final String REPORT_READER = ":reports:read";
+  private static final String SWATCH_APP = "subscriptions";
+  private final ExportApi exportApi;
+  private final RbacService rbacService;
+
+  protected ExportSubscriptionListener(
+      @Qualifier("subscriptionExport") TaskQueueProperties taskQueueProperties,
+      KafkaConsumerRegistry kafkaConsumerRegistry,
+      ConsoleCloudEventParser parser,
+      ExportApi exportApi,
+      RbacService rbacService) {
+    super(taskQueueProperties, kafkaConsumerRegistry);
+    this.parser = parser;
+    this.exportApi = exportApi;
+    this.rbacService = rbacService;
+  }
+
+  @KafkaListener(
+      id = "#{__listener.groupId}",
+      topics = "#{__listener.topic}",
+      containerFactory = "exportListenerContainerFactory")
+  public void receive(String exportEvent) throws ApiException {
+    log.info("New Export event message has been received : {} ", exportEvent);
```

Comment:
Given an export message has a user identity embedded in it, we need to take care not to log the whole message.

Instead, I'd recommend to log at debug level: just `uuid`, `export_request_uuid`, `application`, `format`, and `resource`.